app/code/local/Mage/Core/Model/Session/Abstract/Varien.php

```diff
@@ -116,7 +116,21 @@ public function start($sessionName=null)
             $cookieParams['domain'] = $cookie->getDomain();
         }
 
-        call_user_func_array('session_set_cookie_params', $cookieParams);
+        if (Mage::getStoreConfig(Mage_Core_Model_Cookie::XML_PATH_COOKIE_SAME_SITE)) {
+            $cookieParams['samesite'] = Mage::getStoreConfig(Mage_Core_Model_Cookie::XML_PATH_COOKIE_SAME_SITE);
+        }
+
+        if (isset($cookieParams['samesite']) && PHP_VERSION_ID < 70300) {
+            session_set_cookie_params(
+                $cookieParams['lifetime'],
+                $cookieParams['path'] . '; samesite=' . $cookieParams['samesite'],
+                $cookieParams['domain'],
+                $cookieParams['secure'],
+                $cookieParams['httponly']
+            );
+        } else {
+            call_user_func_array('session_set_cookie_params', $cookieParams);
```

Comment:
This will break if same site is set, because `session_set_cookie_params` only supports the same site param if you call it with an array,